### PR TITLE
track avg work difficulty of active_transactions containers

### PR DESCRIPTION
**Pull request by argakiig**
related to #1336 will satisfy first part

### CODE REVIEW

File: nano/node/node.cpp [hunk 1]
```diff
@@ -3520,6 +3520,7 @@ void nano::active_transactions::stop ()
 bool nano::active_transactions::start (std::shared_ptr<nano::block> block_a, std::function<void(std::shared_ptr<nano::block>)> const & confirmation_action_a)
 {
 	std::lock_guard<std::mutex> lock (mutex);
+	active_total_work += block_a->block_work ();
```
Comment (clemahieu) [inline nano/node/node.cpp]: I think this needs to be adding in the difficulty rather than the random nonce.

File: nano/node/node.cpp [hunk 2]
```diff
@@ -3603,9 +3604,11 @@ void nano::active_transactions::update_difficulty (nano::block const & block_a)
 		uint64_t difficulty;
 		auto error (nano::work_validate (block_a, &difficulty));
 		assert (!error);
+		active_total_work -= existing->difficulty;
 		roots.modify (existing, [difficulty](nano::conflict_info & info_a) {
 			info_a.difficulty = difficulty;
 		});
+		active_total_work += block_a.block_work ();
```
Comment (clemahieu) [inline nano/node/node.cpp]: Difficulty instead of nonce.

File: nano/node/node.hpp [hunk 3]
```diff
@@ -114,6 +114,8 @@ class active_transactions
 	bool vote (std::shared_ptr<nano::vote>, bool = false);
 	// Is the root of this block in the roots container
 	bool active (nano::block const &);
+	//calculate median difficulty;
```
Comment (SergiySW) [inline nano/node/node.hpp]: Average?

File: nano/node/node.cpp [hunk 4]
```diff
@@ -3534,6 +3534,28 @@ bool nano::active_transactions::active (nano::block const & block_a)
...
+		auto size_l (roots.size ());
+		switch (size_l)
+		{
+			case 0:
+				break;
+			case 1:
+				result = roots.get<1> ().begin ()->difficulty;
+				break;
+			default:
+				result = roots.get<1> ().find ((roots.get<1> ().begin ()->difficulty + (--roots.get<1> ().end ())->difficulty) / 2)->difficulty;
```
Comment (SergiySW) [inline nano/node/node.cpp]: Not sure if such estimation makes sense

Comment (argakiig) [reply nano/node/node.cpp]: if not median difficulty, what would you think to be better?

File: nano/node/node.cpp [hunk 5]
```diff
@@ -3534,6 +3534,28 @@ bool nano::active_transactions::active (nano::block const & block_a)
 	return roots.find (nano::uint512_union (block_a.previous (), block_a.root ())) != roots.end ();
 }
 
+uint64_t nano::active_transactions::active_difficulty ()
+{
+	nano::network_params params;
+	uint64_t result (params.publish_threshold);
```
Comment (cryptocode) [inline nano/node/node.cpp]: This can use `node.network_params.publish_threshold` instead of a new params instance.

**Comment by argakiig** [comment]
replaced with #1858

**Pull request closed without merging**